### PR TITLE
Ensure that we don't break sticky on Dangerfiles due to the new syntax

### LINKED ISSUE #235: `fail :sticky` - fails
I bet this is because I added overrides for `def fail(msg);` not `def fail(msg, sticky: true);` in the `Dangerfile` and somewhere else in the plugin work.

See #233

### PR DESCRIPTION
**Pull request by orta**
Fixes #235 and adds an end-to-end test that covers every messaging aspect of the DSL. #trivial

### CODE REVIEW

File: spec/lib/danger/commands/runner_spec.rb [hunk 1]
```diff
@@ -67,6 +67,43 @@ module Command
...
+
+    Dir.mktmpdir do |dir|
+      Dir.chdir dir do
+        `git init`
+        `git remote add origin git@github.com:artsy/eigen.git`
+        File.write("Dangerfile",  %{
+message "Hi"
+warn "OK"
+fail "Not OK"
+markdown "OK [link](http://sure.com)"
```
Comment (KrauseFx) [inline spec/lib/danger/commands/runner_spec.rb]: > Are you a consumer? Pick an island
This is so great cc @neonichu 👍

File: spec/lib/danger/commands/runner_spec.rb [hunk 2]
```diff
@@ -67,6 +67,43 @@ module Command
         end
       end
     end
+    
+      it 'handles an example dangerfile well ' do
+    @git_mock = Danger::GitRepo.new
```
Comment (segiddins) [inline spec/lib/danger/commands/runner_spec.rb]: this indentation looks off

**Comment by neonichu** [comment]
:+1:

File: lib/danger/danger_core/dangerfile.rb [hunk 3]
```diff
@@ -44,13 +44,15 @@ def core_plugin_classes
     # http://ruby-doc.org/core-2.2.3/Kernel.html#method-i-fail
     # However, as we're using using them in the DSL, they won't
     # get method_missing called.
+    
+    # maybe?
 
-    def warn(message)
-      method_missing(:warn, message)
+    def warn(message, sticky: true)
```
Comment (segiddins) [inline lib/danger/danger_core/dangerfile.rb]: Maybe these should take `*args, &blk` and just forward those to `method_missing`?

Comment (orta) [inline lib/danger/danger_core/dangerfile.rb]: great thinking 10/10

**Pull request merged**